daemon/cluster/cluster.go

```diff
@@ -129,6 +112,23 @@ type node struct {
 	conn           *grpc.ClientConn
 	client         swarmapi.ControlClient
 	reconnectDelay time.Duration
+	config         nodeStartConfig
+}
+
+type nodeStartConfig struct {
+	// LocalAddr is this machine's local IP or hostname, if specified.
+	LocalAddr string
+	// RemoteAddr is the address that was given to "swarm join. It is used
+	// to find LocalAddr if necessary.
+	RemoteAddr string
+	// ListenAddr is the address we bind to, including a port.
+	ListenAddr string
+	// AdvertiseAddr is the address other nodes should connect to,
+	// including a port.
+	AdvertiseAddr   string
+	joinAddr        string
+	forceNewCluster bool
+	joinToken       string
```

Comment:
Can you please add a comment that the exported fields must be exported because they are marshalled to json, and the unexported fields must not be exported because they should not be persisted?